src/legacy/ui/public/courier/fetch/components/shard_failure_modal.tsx

```diff
@@ -0,0 +1,136 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiCodeBlock,
+  EuiTabbedContent,
+  EuiCopy,
+  EuiButton,
+  EuiModalBody,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiCallOut,
+} from '@elastic/eui';
+import { ShardFailureTable } from './shard_failure_table';
+import { ResponseWithShardFailure, Request } from './shard_failure_types';
+
+export interface Props {
+  onClose: () => void;
+  request: Request;
+  response: ResponseWithShardFailure;
+  title: string;
+}
+
+export function ShardFailureModal({ request, response, title, onClose }: Props) {
+  if (!response || !response._shards || !Array.isArray(response._shards.failures) || !request) {
+    // this should never ever happen, but just in case
+    const errorTitle = i18n.translate(
+      'common.ui.courier.fetch.shardsFailedModal.componentErrorTitle',
+      {
+        defaultMessage: 'Sorry, there was an error',
+      }
+    );
+    const errorDesc = i18n.translate(
+      'common.ui.courier.fetch.shardsFailedModal.componentErrorDesc',
+      {
+        defaultMessage: 'The ShardFailureModal component received invalid properties',
+      }
+    );
+    return (
+      <EuiCallOut title={errorTitle} color="danger" iconType="alert">
+        {errorDesc}
+      </EuiCallOut>
+    );
+  }
+
+  const requestJSON = JSON.stringify(request, null, 2);
+  const responseJSON = JSON.stringify(response, null, 2);
+  const failures = response._shards.failures;
+
+  const tabs = [
+    {
+      id: 'table',
+      name: i18n.translate('common.ui.courier.fetch.shardsFailedModal.tabHeaderShardFailures', {
+        defaultMessage: 'Shard failures',
+        description: 'Name of the tab displaying shard failures',
+      }),
+      content: <ShardFailureTable failures={failures} />,
+    },
+    {
+      id: 'json-request',
+      name: i18n.translate('common.ui.courier.fetch.shardsFailedModal.tabHeaderRequest', {
+        defaultMessage: 'Request',
+        description: 'Name of the tab displaying the JSON request',
+      }),
+      content: (
+        <EuiCodeBlock language="json" isCopyable className="shardFailureModal__codeblock">
```

Comment:
The `shardFailureModal__codeblock` class is never used, we can remove it. Same for the one below.

Reply:
you've just won the upgrade from Tim to Eagle Eye Tim 👍